client/beadledom-client-example/example-service/src/main/java/com/cerner/beadledom/client/example/PaginatedResourceImpl.java

```diff
@@ -0,0 +1,34 @@
+package com.cerner.beadledom.client.example;
+
+import com.cerner.beadledom.client.example.model.JsonOne;
+import com.cerner.beadledom.jaxrs.GenericResponse;
+import com.cerner.beadledom.jaxrs.GenericResponses;
+import com.cerner.beadledom.pagination.OffsetPaginatedList;
+import com.cerner.beadledom.pagination.parameters.OffsetPaginationParameters;
+
+import com.google.common.collect.Lists;
+import java.util.List;
+import javax.ws.rs.BeanParam;
+
+/**
+ * Service implementation of {@link PaginatedResource}.
+ *
+ * @author Ian Kottman
+ */
+public class PaginatedResourceImpl implements PaginatedResource {
+  @Override
+  public GenericResponse<OffsetPaginatedList<JsonOne>> index(
+      @BeanParam OffsetPaginationParameters paginationParams) {
+    List<JsonOne> items = Lists.newArrayList();
+    for (int i = 0; i < 1000; i++) {
+      items.add(JsonOne.create("One", "Hello World"));
+    }
+
+    OffsetPaginatedList<JsonOne> body = OffsetPaginatedList.<JsonOne>builder()
+        .items(items)
+        .metadata(paginationParams.getLimit(), paginationParams.getOffset(), 1000L, null)
```

Comment:
This could be updated to be `.metadata(1000L)`. We really don't need to pass the limit and offset everywhere since the OffsetPaginatedListLinksWriterInterceptor has access to the UriInfo.

Reply:
I added an another convenience method to the builder for this scenario.